pkg/daemon/image_manager_helper.go

```diff
@@ -0,0 +1,194 @@
+type imageInspection struct {
+	Name          string `json:",omitempty"`
+	Tag           string `json:",omitempty"`
+	Digest        digest.Digest
+	RepoDigests   []string
+	Created       *time.Time
+	DockerVersion string
+	Labels        map[string]string
+	Architecture  string
+	Os            string
+	Layers        []string
+}
+
+// BootedImageInfo stores MCO interested bootec image info
+type BootedImageInfo struct {
+	OSImageURL   string
+	ImageVersion string
+	BaseChecksum string
+}
+
+func podmanInspect(imgURL string) (imgdata *imageInspection, err error) {
+	// Pull the container image if not already available
+	var authArgs []string
+	if _, err := os.Stat(kubeletAuthFile); err == nil {
+		authArgs = append(authArgs, "--authfile", kubeletAuthFile)
+	}
+	args := []string{"pull", "-q"}
+	args = append(args, authArgs...)
+	args = append(args, imgURL)
+	_, err = pivotutils.RunExt(numRetriesNetCommands, "podman", args...)
+	if err != nil {
+		return
+	}
+
+	inspectArgs := []string{"inspect", "--type=image"}
+	inspectArgs = append(inspectArgs, fmt.Sprintf("%s", imgURL))
+	var output []byte
+	output, err = runGetOut("podman", inspectArgs...)
+	if err != nil {
+		return
+	}
+	var imagedataArray []imageInspection
+	err = json.Unmarshal(output, &imagedataArray)
+	if err != nil {
+		err = fmt.Errorf("unmarshaling podman inspect: %w", err)
+		return
+	}
+	imgdata = &imagedataArray[0]
+	return
+
+}
+
+// runGetOut executes a command, logging it, and return the stdout output.
+func runGetOut(command string, args ...string) ([]byte, error) {
+	klog.Infof("Running captured: %s %s", command, strings.Join(args, " "))
+	cmd := exec.Command(command, args...)
+	rawOut, err := cmd.Output()
+	if err != nil {
+		errtext := ""
+		if e, ok := err.(*exec.ExitError); ok {
+			// Trim to max of 256 characters
+			errtext = fmt.Sprintf("\n%s", truncate(string(e.Stderr), 256))
+		}
+		return nil, fmt.Errorf("error running %s %s: %s%s", command, strings.Join(args, " "), err, errtext)
+	}
+	return rawOut, nil
+}
+
+// truncate a string using runes/codepoints as limits.
+// This specifically will avoid breaking a UTF-8 value.
+func truncate(input string, limit int) string {
+	asRunes := []rune(input)
+	l := len(asRunes)
+
+	if limit >= l {
+		return input
+	}
+
+	return fmt.Sprintf("%s [%d more chars]", string(asRunes[:limit]), l-limit)
+}
+
+// useMergedSecrets gives the rpm-ostree / bootc client access to secrets for the internal registry and the global pull
+// secret. It does this by symlinking the merged secrets file into /run/ostree or /etc/ostree. If it fails to find the
+// merged secrets, it will use the default pull secret file instead.
+func useMergedPullSecrets(system imageSystem) error {
+
+	if err := validateImageSystem(system); err != nil {
+		return err
+	}
+
+	// check if merged secret file exists
+	if _, err := os.Stat(imageRegistryAuthFile); err != nil {
```

Comment:
issue: If the error returned by `os.Stat()` is not matched by `errors.Is(err, os.ErrNotExist)`, we just keep going which could be undesirable. Instead, we might want to do something like this:

```golang
if _, err := os.Stat(imageRegistryAuthFile); err != nil {
	if !errors.Is(err, os.ErrNotExist) {
		return fmt.Errorf("could not determine if %s exists: %w", imageRegistryAuthFile, err)
	}

	klog.Errorf("Merged secret file does not exist; defaulting to cluster pull secret")
	return linkAuthFile(system, kubeletAuthFile)
}
```